src/components/Gauge.js

```diff
@@ -0,0 +1,263 @@
+const _isEqual = require('lodash/isEqual');
+const React = require('react');
+const Radium = require('radium');
+const d3 = require('d3');
+
+const StyleConstants = require('../constants/Style');
+
+const Gauge = React.createClass({
+  propTypes: {
+    activeOffset: React.PropTypes.number,
+    arcWidth: React.PropTypes.number,
+    baseArcColor: React.PropTypes.string,
+    chartTotal: React.PropTypes.number,
+    children: React.PropTypes.node,
+    colors: React.PropTypes.array,
+    data: React.PropTypes.array.isRequired,
+    dataPointColors: React.PropTypes.array,
+    dataPointRadius: React.PropTypes.number,
+    dataPoints: React.PropTypes.array,
+    defaultLabelText: React.PropTypes.string,
+    defaultLabelValue: React.PropTypes.string,
+    formatter: React.PropTypes.func,
+    height: React.PropTypes.number,
+    id: React.PropTypes.string,
+    opacity: React.PropTypes.number,
+    padAngle: React.PropTypes.number,
+    showBaseArc: React.PropTypes.bool,
+    showDataLabel: React.PropTypes.bool,
+    width: React.PropTypes.number
+  },
+
+  getDefaultProps () {
+    return {
+      activeOffset: 0,
+      arcWidth: 10,
+      baseArcColor: StyleConstants.Colors.BASE_ARC,
+      colors: [StyleConstants.Colors.PRIMARY].concat(d3.scale.category20().range()),
+      data: [],
+      dataPointColors: [StyleConstants.Colors.CHARCOAL].concat(d3.scale.category20b().range()),
+      dataPointRadius: 5,
+      dataPoints: [],
+      formatter (value) {
+        return value;
+      },
+      height: 150,
+      id: 'gauge',
+      opacity: 1,
+      padAngle: 0.02,
+      showBaseArc: true,
+      showDataLabel: true,
+      width: 150
+    };
+  },
+
+  getInitialState () {
+    return {};
+  },
+
+  componentWillMount () {
+    this._setupD3Functions(this.props);
+  },
+
+  componentWillReceiveProps (newProps) {
+    if (!_isEqual(this.props.data, newProps.data)) {
+      this._setupD3Functions(newProps);
+    }
+  },
+
+  shouldComponentUpdate (nextProps, nextState) {
+    return !_isEqual(this.props, nextProps) || !_isEqual(this.state, nextState);
+  },
+
+  _setupD3Functions (props) {
+    const dataSets = props.data.map(item => {
+      return item.value;
+    });
+    const startAngle = (-135 * (Math.PI / 180));
+    const endAngle = (135 * (Math.PI / 180));
+    const pie = d3.layout.pie().sort(null).padAngle(props.padAngle).endAngle(endAngle);
```

Comment:
Is making the `.sort(null)` call here even required since you're passing `null`?